node/flows.re

```diff
@@ -412,3 +411,40 @@ let request_withdraw_proof = (state, ~hash) =>
   };
 let request_ticket_balance = (state, ~ticket, ~address) =>
   state.Node.protocol.ledger |> Ledger.balance(address, ticket);
+
+let trusted_validators_membership =
+    (~file, ~persist, state, update_state, request)
+    : Lwt.t(result(unit, endpoint_errors('a))) => {
+  open Networking.Trusted_validators_membership_change;
+  let {signature, payload: {address, action} as payload_} = request;
+  let payload_hash =
+    payload_ |> payload_to_yojson |> Yojson.Safe.to_string |> BLAKE2B.hash;
+  if (payload_hash |> Signature.verify(~signature)) {
+    let new_validators =
+      switch (action) {
+      | Add =>
+        Trusted_validators_membership_change.Set.add(
+          {action: Add, address},
+          state.Node.trusted_validator_membership_change,
+        )
+      | Remove =>
+        Trusted_validators_membership_change.Set.remove(
+          {action: Remove, address},
+          state.Node.trusted_validator_membership_change,
+        )
+      };
+    let _: State.t =
+      update_state({
+        ...state,
+        trusted_validator_membership_change: new_validators,
+      });
+    let.await () =
+      persist(
+        new_validators |> Trusted_validators_membership_change.Set.elements,
+        ~file,
+      );
```

Comment:
this persist can happen fully async as it is a function that never fails, use `Lwt.async`, this ensures that the entire flow doesn't use lwt which removes the need for the `handle_request_lwt`.

Reply:
True. Resolving this in the next commit